src/main/java/randoop/main/GenTests.java

```diff
@@ -293,6 +300,11 @@ public boolean handle(String[] args) {
       classNameErrorHandler = new WarnOnBadClassName();
     }
 
+    if (test_add_dependencies) {
```

Comment:
I expected the implementation to be something like

```java
    if (test_add_dependencies) {
      classnames.addAll(getDependentClassnamesFromClassnames(classnames, accessibility));
      classnames.addAll(getDependentClassnamesFromMethodList(accessibility));
      for (int i=1; i<test_add_dependencies_depth; i++) {
        classnames.addAll(getDependentClassnamesFromClassnames(classnames, accessibility));
      }
    }
```

This may be less efficient than other possible implementations, but it permits a much simpler implementation of `getDependentClassnamesFromMethodList` which is agnostic to depth `getDependentClassnamesFromClassnames`.  (You could use your previous, simpler implementations.)

Is there a reason not to prefer an approach like this?

Reply:
Good suggestion! I think it can be improved by adding intermediate variables:
```java
if (test_add_dependencies) {
      classnames.addAll(getDependentClassnamesFromClassnames(classnames, accessibility));
      classnames.addAll(getDependentClassnamesFromMethodList(accessibility));
      Set<@ClassGetName String> searchDependenciesFor = classnames;
      Set<@ClassGetName String> dependencies;
      for (int i = 2; i <= test_add_dependencies_depth; ++i) {
        dependencies = getDependentClassnamesFromClassnames(searchDependenciesFor, accessibility);
        classnames.addAll(dependencies);
        searchDependenciesFor = dependencies;
      }
    }
```
In this case simpler implementation can be used and it is more efficient because we don`t search for dependencies of every class over and over again.

Reply:
This is great.  (It's what I had in mind, but I wanted to paste in a simpler version.)  Thanks for making this change.